### PR TITLE
Update go version and deps

### PR DESCRIPTION
**Pull request by dvail**
Bumps to a more recent version of `go`, one patch release higher than the version declared in stackrox/stackrox.

```
go mod edit -go 1.21.9
go get -u
go mod tidy
```

Testing:
```
go test -v ./resolver
go test -v ./yarn

go build && mv ossls ~/bin
```

Test the output of `make ossls-notice` on stackrox before and after the change:
```
$ make ossls-notice
$ diff -rq image/rhel/THIRD_PARTY_NOTICES image/rhel/THIRD_PARTY_NOTICES.pre-mod
Files image/rhel/THIRD_PARTY_NOTICES/manifest.csv and image/rhel/THIRD_PARTY_NOTICES.pre-mod/manifest.csv differ
```

Manual inspection of both manifest.csv files shows a slight change in the ordering, but otherwise contents is the same. The updated ossls seems to handle version numbers in ascending order now, e.g. with `difft`:
```
2340 wrap-ansi,7.0.0,./wrap-ansi7.0.0                       ....
2341 wrap-ansi,6.2.0,./wrap-ansi6.2.0                       2340 wrap-ansi,6.2.0,./wrap-ansi6.2.0
....                                                        2341 wrap-ansi,7.0.0,./wrap-ansi7.0.0
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,15 +1,23 @@
 module github.com/stackrox/ossls
 
-go 1.14
+go 1.21.9
```
Comment (RTann) [inline go.mod:3]:
```suggestion
go 1.21

toolchain go1.21.9
```

This way users of this (if this were used as a library) are not required to use go1.21.9+, but rather any go1.21.0+. Setting `toolchain` means that go version is required when this is run as the main module.

**Comment by RTann** [comment]
This seems unrelated to the other PR you have. I think this should just be based on the master/main branch and may be merged before that other PR

File: go.mod [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+	github.com/BurntSushi/toml v1.4.0
+	github.com/fatih/color v1.17.0
 	github.com/ghodss/yaml v1.0.0
-	github.com/mattn/go-colorable v0.0.9 // indirect
-	github.com/mattn/go-isatty v0.0.8 // indirect
-	github.com/pkg/errors v0.8.1
-	github.com/spf13/cobra v0.0.4
+	github.com/pkg/errors v0.9.1
+	github.com/spf13/cobra v1.8.1
 	github.com/stretchr/testify v1.3.0
```
Comment (RTann) [inline go.mod:13]: I guess while you're at it, want to bump this to `v1.9.0`?

Comment (dvail) [reply go.mod]: Might as well :)

**Review by RTann — APPROVED** [review]

**Pull request merged**